src/libexpr/primops.cc

```diff
@@ -519,14 +519,27 @@ void prim_valueSize(EvalState & state, const Pos & pos, Value * * args, Value &
  *************************************************************/
 
 
+/* Add a wrapper around the derivation primop that computes the
+   `drvPath', `outPath', `inputSrcs' and `inputDrvs' attributes lazily. */
+static void prim_derivation(EvalState & state, const Pos & pos, Value * * args, Value & v)
+{
+    Value fun;
+    state.evalFile(state.sDerivationNix, fun);
+    state.forceFunction(fun, pos);
+    mkApp(v, fun, *args[0]);
+    state.forceAttrs(v, pos);
+}
```

Comment:
Although this wrapper is a little silly from a mechanical perspective, if we register it with a docstring that would solve a little UX problem that we have. (Could be an independent PR as well)